### PR TITLE
Throw an error if dashes are used

### PR DESCRIPTION
**Pull request by Martinn1996**
Since version 2 of MessageFormat, dashes, "-", can no longer be used. 
So if we detect a dash, we will throw an error.

### CODE REVIEW

**Review by bwrrp — CHANGES_REQUESTED** [review]
Looks good. Instead of translating the string to underscores, let's just throw directly from the parser when dashes are matched in the wrong context, and catch + re-throw in Message.js with an error that includes the full message.

**Review by DrRataplan — APPROVED** [review]
Looks great! If you can edit the parser to just throw instead of returning a '_', that would be awesome.

File: src/Message.js [hunk 1]
```diff
@@ -38,15 +38,11 @@ module.exports = class Message {
 	 * @param  {Object|Object[]}  [props.metadata]
 	 */
 	constructor (props) {
-		if (props.original !== parser.parse(props.original)) {
-			throw new Error('Cannot use dashes (-) in ' + props.original);
-		}
+		parser.parse(props.original);
```
Comment (bwrrp) [inline src/Message.js]: Let's catch and re-throw the error in both cases in order to include the full message: `Dashes are not allowed in identifiers, in message "${props.original}"`

File: src/parser.pegjs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+select = '{' _ id _ ',' _ ('select' { if (options.strictNumberSign) { inPlural = false; } }) _ ',' _ (cases:selectCase+) _ '}'
+
+plural = '{' _ id _ ',' _ (('plural'/'selectordinal') { inPlural = true; } ) _ ',' _ offset? (pluralCase+ ) _ '}'
+
+function = '{' _ id _ ',' _ (id { if (options.strictNumberSign) { inPlural = false; } }) _ functionParam? '}'
+
+// not Pattern_Syntax or Pattern_White_Space
+// TODO: Add additional replaceable characters here:
+id = x:('-' { throw new Error("Dashes are now allowed (-)")} / $[^\u0009-\u000d \u0085\u200e\u200f\u2028\u2029\u0021-\u002f\u003a-\u0040\u005b-\u005e\u0060\u007b-\u007e\u00a1-\u00a7\u00a9\u00ab\u00ac\u00ae\u00b0\u00b1\u00b6\u00bb\u00bf\u00d7\u00f7\u2010-\u2027\u2030-\u203e\u2041-\u2053\u2055-\u205e\u2190-\u245f\u2500-\u2775\u2794-\u2bff\u2e00-\u2e7f\u3001-\u3003\u3008-\u3020\u3030\ufd3e\ufd3f\ufe45\ufe46])+ {return x.join('')}
```
Comment (DrRataplan) [inline src/parser.pegjs]: are no*t*

Comment (bwrrp) [inline src/parser.pegjs]: I think only this code block is still necessary, and the rest can be removed?

**Review by bwrrp — CHANGES_REQUESTED** [review]

**Pull request merged**